### DISCUSSION TITLE
Can the Zemismart Tuya Zigbee WiFi 3 Phase Energy Meter be supported?

### DISCUSSION THREAD

**Discussion by vlacemeister** [make-all/tuya-local#1687] [Q&A]
![device](https://github.com/make-all/tuya-local/assets/144171904/46cae2d5-4dbd-411b-a4f0-985de11d0fb2)

Hello all! (First time asking something here)

I bought the Zemismart Tuya Zigbee WiFi 3 Phase Energy Meter (https://vi.aliexpress.com/item/1005005879995196.html).
- LocalTuya not installed.
- Only protocol 3.4 works.
- Tuya category: dlq.

![adding-device](https://github.com/make-all/tuya-local/assets/144171904/505e8920-750e-4157-b286-75030532a20f)

When I try do add it to Tuya-Local I get 2 different results, apart from "Unable to connect..." under other protocols.

Sometimes I get the "Sorry, there is no support for this device." message. 

![sorry](https://github.com/make-all/tuya-local/assets/144171904/51bc36f0-ca05-4b82-beef-827fe227fc74)

Sometimes I get a different set of devices to choose from.

![other-device](https://github.com/make-all/tuya-local/assets/144171904/e2461b45-36ab-4f68-80b0-13a1e360166e)

Oddly enough, the day before the release of HA 2024.2 it worked for a while until HA got updated.

Can this model be supported?

Thanks

{
        "modelId": "000002yenr",
        "services": [{
                "actions": [],
                "code": "",
                "description": "",
                "events": [],
                "name": "默认服务",
                "properties": [{
                        "abilityId": 1,
                        "accessMode": "ro",
                        "code": "total_forward_energy",
                        "description": "",
                        "extensions": {
                            "iconName": "icon-dp_lightning",
                            "attribute": "128"
                        },
                        "name": "累计正向有功电能",
                        "typeSpec": {
                            "max": 99999999,
                            "min": 0,
                            "scale": 2,
                            "step": 1,
                            "type": "value",
                            "typeDefaultValue": 0,
                            "unit": "kW·h"
                        }
                    }, {
                        "abilityId": 2,
                        "accessMode": "ro",
                        "code": "cur_neutral",
                        "description": "",
                        "extensions": {
                            "iconName": "icon-power1",
                            "attribute": "128"
                        },
                        "name": "累计反向有功电能",
                        "typeSpec": {
                            "max": 99999999,
                            "min": 0,
                            "scale": 2,
                            "step": 1,
                            "type": "value",
                            "typeDefaultValue": 0,
                            "unit": "kW.h"
                        }
                    }, {
                        "abilityId": 6,
                        "accessMode": "ro",
                        "code": "phase_a",
                        "description": "1，A相电压，电流及功率\\n2，大端模式，HEX格式，共8个字节\\n3，单位精度：电压，2字节，单位0.1V。电流，3字节，单位 0.001A 。A相有功功率，3字节，单位0.0001kW\\n4，报文格式\\n举例：08 80 00 03 E8 00 27 10 表示A相217.6V，A相电流1.000A，A相功率10.000KW\\n5，通信逻辑：\\n1）用户进面板，主动查询。用户进入面板，面板马上下发0x08到电表，电表上报数据到云平台，面板拿到数据，然后展示。 注：最新WIFI支持。   \\n2）电表基于一定周期上报数据。建议：在WIFI模式时，15秒上报一次。NB模式时，1个小时上报一次。",
                        "extensions": {
                            "iconName": "icon-dp_lightning",
                            "attribute": "128"
                        },
                        "name": "A相电压，电流及功率",
                        "typeSpec": {
                            "maxlen": 128,
                            "type": "raw"
                        }
                    }, {
                        "abilityId": 7,
                        "accessMode": "ro",
                        "code": "phase_b",
                        "description": "1，B相电压，电流及功率\\n2，大端模式，HEX格式，共8个字节\\n3，单位精度：电压，2字节，单位0.1V。电流，3字节，单位 0.001A 。B相有功功率，3字节，单位0.0001kW\\n4，报文格式\\n举例：08 80 00 03 E8 00 27 10 表示B相217.6V，B相电流1.000A，B相功率10.000KW\\n5，通信逻辑：\\n1）用户进面板，主动查询。用户进入面板，面板马上下发0x00到电表，电表上报数据到云平台，面板拿到数据，然后展示。    \\n2）电表基于一定周期上报数据。建议：在WIFI模式时，15秒上报一次。NB模式时，1个小时上报一次。",
                        "extensions": {
                            "iconName": "icon-dp_lightning",
                            "attribute": "128"
                        },
                        "name": "B相电压，电流及功率",
                        "typeSpec": {
                            "maxlen": 128,
                            "type": "raw"
                        }
                    }, {
                        "abilityId": 8,
                        "accessMode": "ro",
                        "code": "phase_c",
                        "description": "1，C相电压，电流及功率\\n2，大端模式，HEX格式，共8个字节\\n3，单位精度：电压，2字节，单位0.1V。电流，3字节，单位 0.001A 。C相有功功率，3字节，单位0.0001kW\\n4，报文格式\\n举例：08 80 00 03 E8 00 27 10 表示C相217.6V，C相电流1.000A，C相功率10.000KW\\n5，通信逻辑：\\n1）用户进面板，主动查询。用户进入面板，面板马上下发0x08到电表，电表上报数据到云平台，面板拿到数据，然后展示。    \\n2）电表基于一定周期上报数据。建议：在WIFI模式时，15秒上报一次。NB模式时，1个小时上报一次。",
                        "extensions": {
                            "iconName": "icon-dp_lightning",
                            "attribute": "128"
                        },
                        "name": "C相电压，电流及功率",
                        "typeSpec": {
                            "maxlen": 128,
                            "type": "raw"
                        }
                    }, {
                        "abilityId": 9,
                        "accessMode": "ro",
                        "code": "fault",
                        "description": "1，告警\\n2，大端模式，HEX,4字节\\n3，报文格式\\n按照完整4个字节,对应的告警位上报告警；bit位，1表示有告警，0表示无告警；如果没用的告警位，补0。高位无定义，也补0；\\n举例：0X00 00 00 09,表示当前发生了漏电自检不正常告警和短路告警,无其它告警。\\n4，通信逻辑\\n只要有1个告警发生，就完整4个字节上报给面板；只要有1个告警关闭，就完整4个字节上报给面板，更新面板显示。",
                        "extensions": {
                            "iconName": "icon-baojing",
                            "attribute": "128"
                        },
                        "name": "故障告警",
                        "typeSpec": {
                            "label": ["short_circuit_alarm", "surge_alarm", "overload_alarm", "leakagecurr_alarm", "temp_dif_fault", "fire_alarm", "high_power_alarm", "self_test_alarm", "ov_cr", "unbalance_alarm", "ov_vol", "undervoltage_alarm", "miss_phase_alarm", "outage_alarm", "magnetism_alarm", "credit_alarm", "no_balance_alarm"],
                            "maxlen": 17,
                            "type": "bitmap",
                            "typeDefaultValue": 0
                        }
                    }, {
                        "abilityId": 11,
                        "accessMode": "rw",
                        "code": "switch_prepayment",
                        "description": "",
                        "extensions": {
                            "iconName": "icon-dp_money",
                            "attribute": "1152"
                        },
                        "name": "预付费功能开关",
                        "typeSpec": {
                            "type": "bool",
                            "typeDefaultValue": false
                        }
                    }, {
                        "abilityId": 12,
                        "accessMode": "rw",
                        "code": "clear_energy",
                        "description": "",
                        "extensions": {
                            "iconName": "icon-dp_lightning",
                            "attribute": "1152"
                        },
                        "name": "余额清零",
                        "typeSpec": {
                            "type": "bool",
                            "typeDefaultValue": false
                        }
                    }, {
                        "abilityId": 13,
                        "accessMode": "ro",
                        "code": "balance_energy",
                        "description": "",
                        "extensions": {
                            "iconName": "icon-dp_lightning",
                            "attribute": "1152"
                        },
                        "name": "余额",
                        "typeSpec": {
                            "max": 99999999,
                            "min": 0,
                            "scale": 2,
                            "step": 1,
                            "type": "value",
                            "typeDefaultValue": 0,
                            "unit": "kW·h"
                        }
                    }, {
                        "abilityId": 14,
                        "accessMode": "rw",
                        "code": "charge_energy",
                        "description": "",
                        "extensions": {
                            "iconName": "icon-dp_money",
                            "attribute": "1152"
                        },
                        "name": "电量充值",
                        "typeSpec": {
                            "max": 999999,
                            "min": 0,
                            "scale": 2,
                            "step": 1,
                            "type": "value",
                            "typeDefaultValue": 0,
                            "unit": "kW·h"
                        }
                    }, {
                        "abilityId": 17,
                        "accessMode": "rw",
                        "code": "alarm_set_1",
                        "description": "1，告警阈值使能控制及设置2  \\n2，HEX格式，大端模式, 总告警的数量*4字节= 设置告警总长度。\\n3，报文格式\\n每个告警由4个字节组成，第1个字节（表示该告警存在）；第2个字节（发生此告警时，断路器控制模式（0X01拉闸，0X00不动作只告警）；第3、4字节：设置告警阈值\\n举例：0X01 01 00 00 03 01 00 14,表示该断路器支持短路告警，发生该告警时，自动跳闸，阀值补0。支持过载告警，发生该告警时，拉闸动作，动作阀值20KW。无其它告警。\\n4，通信逻辑\\nA，配网成功后，按照上面的格式，主动上报给云平台，面板知道哪些告警是断路器当前支持的。设置不成功，重新配网再上报。\\nB，断路器上电时，也上报一次。\\nC，面板根据协议下发告警设置。",
                        "extensions": {
                            "iconName": "icon-dp_warming",
                            "attribute": "128"
                        },
                        "name": "告警设置1",
                        "typeSpec": {
                            "maxlen": 128,
                            "type": "raw"
                        }
                    }, {
                        "abilityId": 18,
                        "accessMode": "rw",
                        "code": "alarm_set_2",
                        "description": "1，告警阈值使能控制及设置2  \\n2，HEX格式，大端模式\\n3，报文格式\\n每个告警由4个字节组成，第1个字节（表示该告警存在）；第2个字节（发生此告警时，断路器控制模式（0X01拉闸，0X00不动作只告警）；第3、4字节：设置告警阈值\\n举例：0X01 01 00 3C 03 01 00 F5,表示该断路器支持过流告警，发生该告警时，拉闸动作，动作阀值60A。支持过压告警，发生该告警时，拉闸动作，动作阀值245V\\n4，通信逻辑\\nA，配网成功后，按照上面的格式，主动上报给云平台，面板知道哪些告警是断路器当前支持的。设置不成功，重新配网再上报。\\nB，断路器上电时，也上报一次。\\nC，面板根据协议下发告警设置。",
                        "extensions": {
                            "iconName": "icon-dp_warming",
                            "attribute": "128"
                        },
                        "name": "告警设置2",
                        "typeSpec": {
                            "maxlen": 128,
                            "type": "raw"
                        }
                    }
                ]
            }
        ]
    }

**Comment by make-all** [comment]
This should match yagusmart_3pn_energymeter.  If not in the current release, in the next release after a compatibility improvement committed today.
